Library/Homebrew/diagnostic.rb

```diff
@@ -147,6 +147,27 @@ def examine_git_origin(repository_path, desired_origin)
         end
       end
 
+      def broken_tap_msg(tap)
+        <<~EOS
+          #{tap.full_name} was not tapped properly.
+          To fix:
+            rm -rf "#{tap.path}"
+            brew tap #{tap.name}
+        EOS
+      end
+
+      def broken_tap(tap)
+        return unless Utils::Git.available?
+        return unless HOMEBREW_REPOSITORY.git?
+
+        return broken_tap_msg(tap) if tap.remote.blank?
+
+        tap_head = tap.git_head
+        return broken_tap_msg(tap) if tap_head.blank?
+
+        return broken_tap_msg(tap) if tap_head == HOMEBREW_REPOSITORY.git_head
+      end
```

Comment:
```suggestion
      def broken_tap(tap)
        return unless Utils::Git.available?
        return unless HOMEBREW_REPOSITORY.git?

        message = <<~EOS
          #{tap.full_name} was not tapped properly! Run:
            rm -rf "#{tap.path}"
            brew tap #{tap.name}
        EOS

        return message if tap.remote.blank?

        tap_head = tap.git_head
        return message if tap_head.blank?
        return if tap_head != HOMEBREW_REPOSITORY.git_head
        
        message
      end
```

Reply:
Argh, sometimes I miss the obvious. :rofl: